index.js

```diff
@@ -316,6 +316,10 @@ function HttpRequest(options) {
 			}
 		};
 	}
+
+	if ('onerror' in xhr && !xhr.onreadystatechange) {
+		xhr.onerror = onLoad;
```

Comment:
This thing really had to support IE9 in the past. So that may have been why, indeed. There's also PhantomJS I can imagine being "special" here. I don't remember any specific reasons beyond these possible issues.